update Compose release notes page for v2.20.1 and v2.20.2 (#17764)
docker/docs: compose/release-notes.md

For more detailed information, see the [release notes in the Compose repo](https://github.com/docker/compose/releases/).

## 2.20.2
{% include release-date.html date="2023-07-19" %}

### Bug fixes and enhancements

> - Add support of `depends_on.required` attribute

Reviewer: ```suggestion
- Added support for the `depends_on.required` attribute.
```

> - Fixed an issue about build trying to push unnamed service images

Reviewer: ```suggestion
- Fixed an issue where build tries to push unnamed service images.
```

> - Fixed a bug with target secret path on Windows

Reviewer 2: ```suggestion
- Fixed a bug with target secret path on Windows
- Fixed a bug resolving build context path for services using `extends.file`
```

That fix comes from `compose-go` but was reported as docker/for-mac#6912, so I think might be worth an explicit callout.

> - Fixed a bug with target secret path on Windows

Reviewer: ```suggestion
- Fixed a bug meant the target secret path on Windows was not checked.
```

For more detailed information, see the [release notes in the Compose repo](https://github.com/docker/compose/releases/).

## 2.20.2
{% include release-date.html date="2023-07-19" %}

### Bug fixes and enhancements
- Added support for the `depends_on.required` attribute.
- Fixed an issue where build tries to push unnamed service images.

> - Fixed a bug with target secret path on Windows

Reviewer: ```suggestion
- Fixed a bug which meant the target secret path on Windows was not checked.
```

> - Fixed a bug resolving build context path for services using `extends.file`

Reviewer: ```suggestion
- Fixed a bug resolving build context path for services using `extends.file`.
```
